packages/@aws-cdk/aws-ecs/lib/queue-worker-service-base.ts

```diff
@@ -0,0 +1,180 @@
+import autoscaling = require('@aws-cdk/aws-applicationautoscaling');
+import sqs = require('@aws-cdk/aws-sqs');
+import { Queue } from '@aws-cdk/aws-sqs';
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base/base-service';
+import { ICluster } from './cluster';
+import { ContainerImage } from './container-image';
+import { AwsLogDriver } from './log-drivers/aws-log-driver';
+import { LogDriver } from './log-drivers/log-driver';
+
+/**
+ * Properties to define a Query Worker service
+ */
+export interface QueueWorkerServiceBaseProps {
+  /**
+   * Cluster where service will be deployed
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The image to start.
+   */
+  readonly image: ContainerImage;
+
+  /**
+   * The CMD value to pass to the container. A string with commands delimited by commas.
+   *
+   * @default none
+   */
+  readonly command?: string;
+
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  readonly desiredTaskCount?: number;
+
+  /**
+   * Flag to indicate whether to enable logging
+   *
+   * @default true
+   */
+  readonly enableLogging?: boolean;
+
+  /**
+   * The environment variables to pass to the container.
+   *
+   * @default none
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * A name for the queue.
+   *
+   * If specified and this is a FIFO queue, must end in the string '.fifo'.
+   *
+   * @default 'EcsWorkerServiceQueue'
```

Comment:
I think you would want to have this name autogenerated if absent, no?

Reply:
Updated.